src/fsharp/DependencyManager.Integration.fs

```diff
@@ -84,35 +85,66 @@ type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyIn
+            let resolveMethodEx = ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
+            Some (fun () -> new ReflectionDependencyManagerProvider(theType, nameProperty, keyProperty, resolveMethod, resolveMethodEx, outputDir) :> IDependencyManagerProvider)
 
     interface IDependencyManagerProvider with
 
         member __.Name     = instance |> nameProperty
 
         member __.Key      = instance |> keyProperty
-        member this.ResolveDependencies(scriptDir, packageManagerTextLines, tfm) =
+
+        member this.ResolveDependencies(scriptDir, mainScriptName, scriptName, scriptExt, packageManagerTextLines, tfm) =
 
             let key = (this :> IDependencyManagerProvider).Key
             let triggerEvent (evt: Event<string * string>) =
                 for prLine in packageManagerTextLines do
                     evt.Trigger(key, prLine)
             triggerEvent dependencyAddingEvent
-            let arguments = [| box scriptDir; box packageManagerTextLines; box tfm |]
-            let succeeded, generatedScripts, additionalIncludeFolders = resolveDeps.Invoke(instance, arguments) :?> _
+
+
+            // The ResolveDependencies method, has two signatures, the original signaature in the variable resolveDeps and the updated signature resoveDepsEx
+            // The resolve method can return values in two different tuples:
+            //     (bool * string list * string list * string list)
+            //     (bool * string list * string list)
+            //
+            // We use reflection to get the correct method and to determine what we got back.
+            //
+            let method, arguments =
+                if resolveDepsEx.IsSome then
+                    resolveDepsEx, [| box scriptExt; box packageManagerTextLines; box tfm |]
+                elif resolveDeps.IsSome then
+                    resolveDeps, [| box scriptDir; box mainScriptName; box scriptName; box packageManagerTextLines; box tfm |]
+                else
+                    None, [||]
+
+            let succeeded, references, generatedScripts, additionalIncludeFolders =
+                let empty = List.empty<string>
+                let result =
+                    match method with
+                    | Some m -> m.Invoke(instance, arguments) :?> _
+                    | None -> false, empty, empty, empty
+
+                // Verify the number of arguments returned in the tuple returned by resolvedependencies, it can be:
+                //     4 - (bool * string list * string list * string list)
+                //     3 - (bool * string list * string list)
+                let tupleFields = result |> FSharpValue.GetTupleFields
```

Comment:
Could `FSharpValue.GetTupleFields` ever throw?

Reply:
The errors will propagate quite nicely as compiler errors.  Although perhaps we could wrap it in a your package manger returned invalid arguments.